src/Pos/Wallet/Web/ClientTypes.hs

```diff
@@ -161,21 +161,22 @@ convertTxOutputs :: [TxOut] -> [(CId w, CCoin)]
 convertTxOutputs = map (addressToCId . txOutAddress &&& mkCCoin . txOutValue)
 
 -- | Get all addresses of source account of given transaction.
-getLocalAccountAddrs
-    :: [CWAddressMeta]  -- ^ all addresses in wallet
-    -> [CId Addr]       -- ^ Input addresses of transaction
-    -> Either Text [CId Addr]
-getLocalAccountAddrs walAddrMetas inputAddrs = do
+getChangeAccountAddrs
+    :: [CWAddressMeta]   -- ^ All addresses in wallet
+    -> [CId Addr]        -- ^ Input addresses of transaction
+    -> Either Text (Maybe [CId Addr])
+                         -- ^ `Just` change addrs if the wallet is source of
+                         --   transaction, `Nothing` otherwise
+getChangeAccountAddrs walAddrMetas inputAddrs = do
     someInputAddr <-
         head inputAddrs `whenNothing`
         throwError "No input addresses in transaction"
-    someSrcAddrMeta <-
-        find ((== someInputAddr) . cwamId) walAddrMetas `whenNothing`
-        throwError "Address doesn't belong to any wallet"
-    let srcAccount =
-            addrMetaToAccount someSrcAddrMeta
-    return $
-        map cwamId $ filter ((srcAccount ==) . addrMetaToAccount) walAddrMetas
+    return $ do
```

Comment:
It might be just me, but it takes some mental effort to figure out that the monad here is `Maybe`. Perhaps you could add a comment to clarify that?